src/parser/index.ts

```diff
@@ -124,3 +127,33 @@ async function parseSection(ctx: ParserContext, src: Source): Promise<WITSection
 
     return sections;
 }
+
+async function parseSectionComponent(
+    ctx: ParserContext,
+    src: Source,
+    size: number
+): Promise<ComponentTypeComponent[]> {
+    const bytes = await src.readExact(size);
+    const src2 = newSource(bytes);
+
+    await checkPreamble(src2);
+    const model: ComponentTypeDeclaration[] = [];
+    for (; ;) {
+        const sections = await parseSection(ctx, src2);
+        if (sections === null) {
+            break;
+        }
+        for (const s of sections) {
+            switch (s.tag) {
+                default:
+                    console.log(JSON.stringify(s));
+                    //throw new Error(`Unexpected section ${s.tag}`);
+                    break;
+            }
+        }
```

Comment:
@ilonatommy This is the section that needs to transform the `WITSection` instances into the correct `ComponentTypeDeclaration` type based on the `s.tag`. This parses correctly, but the translation was more than I could handle today. Look in `hello.ts` for how the they should be translated.